zio-cli/shared/src/main/scala/zio/cli/ConfigFileArgs.scala

```diff
@@ -0,0 +1,9 @@
+package zio.cli
+
+import zio._
+import java.io.IOException
+
+trait ConfigFilePlatformSpecific {
+  def findPathsOfCliConfigFiles(topLevelCommand: String): Task[List[String]]
+  def loadOptionsFromConfigFiles(topLevelCommand: String): ZIO[Any, IOException, List[String]]
+}
```

Comment:
Personal opinion:  
I think it would be nicer if this passed via something `~like`
```scala
trait ConfigFileArgs
object ConfigFileArgs {
  val default: ConfigFileArgs // platform specific
}
```
```scala
sealed trait CliApp[-R, +E, +A] { self =>
  def runWithFileArgs(args: List[String]): ZIO[R & ConfigFileArgs, CliError[E], Option[A]]
  final def run(args: List[String]): ZIO[R, CliError[E], Option[A]] = 
    runWithFileArgs(args).provideSomeLayer(ZLayer.succeed(ConfigFileArgs.default))
```